### DISCUSSION TITLE
Fundamentos do c#

### DISCUSSION THREAD

**Discussion by daniloandersgomes** [balta-io/forum#652] [Tira Dúvidas] [answered]
Não consigo pesquisar por .NET Core para iniciar uma pasta vscode para usar o debugger, a pesquisa não dá resultado nenhum

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @daniloandersgomes, como vai?

Qual erro ocorre? Tem algum print para ajudar na visualização do erro?

**Reply by daniloandersgomes** [reply]
Não ocorre um erro em especifico, mas a pesquisa não dá o mesmo resultado que deu na sua aula, e a pasta vscode não é criada... Tambem não consegui adicionar os demais icones na parte esquerda do app, junto ao icone das extensões, debug, explorer, search, etc.


![Captura de tela 2024-08-08 070122](https://github.com/user-attachments/assets/a743b4ab-0f65-493f-8256-096f52324776)

**Reply by andrebaltieri** [reply]
Pelo que parece aqui ele nem criou a pasta Properties ou a .vscode... por isso não aparece nenhuma opção de debug.
Tenta reinstalar as extensões do C#, incluindo a nova C# DevKit por favor.

Se o erro persistir, tenta reinstalar o VS Code! As vezes é algo no cache do Windows...

**Comment by AmynadabeLRocha** [comment]
pergunta???
posso seguir o curso utilizando o modelo de programa novo? não atrapalha o aprendizado ?? 
![estilo de programa novo ](https://github.com/user-attachments/assets/aaadd391-cac2-43f1-a230-7c8ef8a0aea9)

**Reply by kauepiovan** [reply]
Você pode, este é um recurso mais recente do .NET chamado de [Top Level Statements](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/top-level-statementsl), porem eu recomendo não utilizar ele na parte de fundamentos do C#, justamente para você ficar na mesma pagina do Balta.
Agora no .NET pra você criar uma aplicação do tipo console com o método Main no Program.cs você tem que utilizar a seguinte linha de comando no terminal.
`dotnet new console -o NomeDaAplicacao --use-program-main`
criando com este comando seu código inicial vai ficar igual o do Balta neste modulo de fundamentos do C#.

**Reply by andrebaltieri** [reply]
No caso, para criar o programa com a mesma estrutura que uso no curso é só usar o `--use-program-main` na hora de criar o projeto:
```
dotnet new console -o MeuApp --use-program-main
```

Caso esteja na versão mais antiga do .NET (3.1 e/ou 5), ele já cria igual do curso... se for mais recente, ai usa a linha acima.

Outro ponto é que o .NET roda lado-a-lado, isto significa que você pode instalar diversas versões do .NET na sua máquina e uma não impacta outra.

Para criar um projeto em uma versão específica, você usa o `-f`:
```
dotnet new console -o MeuApp -f net3.1
dotnet new console -o MeuApp -f net5
dotnet new console -o MeuApp -f net6
```

**Reply by AmynadabeLRocha** [reply]
Muito obrigado ! Então sempre antes de criar uma aplicação eu utilizar este comando no terminal é o melhor a fazer certo ? 
Muito obrigado ! Um forte abraço! 

> dotnet new console -o MeuApp --use-program-main

**Reply by AmynadabeLRocha** [reply]
> No caso, para criar o programa com a mesma estrutura que uso no curso é só usar o `--use-program-main`na hora de criar o projeto:
> 
> ```
> dotnet new console -o MeuApp --use-program-main
> ```
> 
> Caso esteja na versão mais antiga do .NET (3.1 e/ou 5), ele já criou igual ao curso... se for mais recente, ai usa a linha acima.
> 
> Outro ponto é que o .NET roda lado a lado, isso significa que você pode instalar diversas versões do .NET em sua máquina e uma não impacta outra.
> 
> Para criar um projeto em uma versão específica, você usa o `-f`:
> 
> ```
> dotnet new console -o MeuApp -f net3.1
> dotnet new console -o MeuApp -f net5
> dotnet new console -o MeuApp -f net6
> ```

![NET](https://github.com/user-attachments/assets/34a4166c-85b5-4453-8a1f-d64e74e1005a)


Esse foi o resultado ... a Linha  using System não aparece.

**Reply by andrebaltieri** [reply]
No caso, a linha `using System` não aparece pois a versão do .NET que você está usando já faz uso do recurso chamado **Implicit Using**, no qual ele já entende que o `System` é um **namespace** tão popular, que não se cria nada sem ele.

Desta forma, a partir da versão 5 do .NET se não me engano, não é mais preciso a linha `using System`.

Esta configura fica definida no seu arquivo `.csproj` através da tag `ImplicitUsing`.
